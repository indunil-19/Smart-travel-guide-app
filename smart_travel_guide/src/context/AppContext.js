import React, {createContext, useReducer} from 'react';
import { initialState, reducer as AppReducer } from '../reducer/AppReducer';

export const AppContext =createContext()

export const AppProvider=({children})=>{
    const [state, dispatch]= useReducer(AppReducer, initialState)

    return(
        <AppContext.Provider  value={{state,dispatch}}>
                {children}
        </AppContext.Provider>
    )
}