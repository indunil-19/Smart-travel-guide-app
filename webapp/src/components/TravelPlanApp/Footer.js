import { Flex, HStack, VStack,Text ,Heading,Divider} from "@chakra-ui/layout"
import { Link } from "react-router-dom"

export const Footer=()=>{
    return(
        <>
        <Flex width="full" bg="black"  mt="165px" color="white" justifyContent="space-around" p={10}  >
             <VStack>
                <HStack spacing={10}>
                <VStack alignItems="start" spacing={8}>

                     
                     
                     <Heading as="h3" size="lg">
                        About Us
                    </Heading>
                 
                    

                    <Heading as="h3" size="lg">
                        Contact Us
                    </Heading>
                  
                </VStack>

                <Divider orientation="vertical"   variant="solid" p={3}  />

                <VStack>
                     
                     <>

                     <iframe src="https://www.facebook.com/plugins/page.php?href=https%3A%2F%2Fwww.facebook.com%2FWonderDiscoveryTravelSriLanka%2F&tabs=messages&width=340&height=500&small_header=false&adapt_container_width=true&hide_cover=false&show_facepile=true&appId" width="340" height="500" style={{border:"none",overflow:"hidden"}} scrolling="no" frameborder="0" allowfullscreen="true" allow="autoplay; clipboard-write; encrypted-media; picture-in-picture; web-share"></iframe>                     </>
                </VStack>
                </HStack>
                       <Divider  orientation="horizontal"   variant="solid" p={3}  />
                </VStack>
               
        </Flex>
       


        </>
    )
}