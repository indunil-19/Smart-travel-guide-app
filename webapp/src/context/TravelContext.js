import React, {createContext, useReducer} from 'react';
import { initialState, Travelreducer } from '../reducer/TravelReducer';

const TravelContext =createContext()

const TravelProvider=({children})=>{
    const [state, dispatch]= useReducer(Travelreducer, initialState)

    return(
        <TravelContext.Provider  value={{state,dispatch}}>
                {children}
        </TravelContext.Provider>
    )
}

export  {TravelContext,  TravelProvider}
 