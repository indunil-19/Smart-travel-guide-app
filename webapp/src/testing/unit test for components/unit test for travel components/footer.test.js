import React from "react"
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { MemoryRouter,Router } from "react-router-dom";
import { TravelProvider } from "../../../context/TravelContext";
import "@testing-library/jest-dom/extend-expect";
import 'whatwg-fetch'
import Carousal from "../../../components/TravelPlanApp/carousal";
import { getByTestId, waitFor } from "@testing-library/dom";
import Footer from "../../../components/TravelPlanApp/Footer";

describe('footer component test', () => {
    
    it("should match snapshot",  () => {
        const tree = renderer.create(<MemoryRouter><TravelProvider><Footer/></TravelProvider></MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });

    it("shoud render without crashing", ()=>{
        const div = document.createElement("div");
        ReactDOM.render(<MemoryRouter><TravelProvider><Footer  /></TravelProvider></MemoryRouter>,div)
       
       
    })
})
