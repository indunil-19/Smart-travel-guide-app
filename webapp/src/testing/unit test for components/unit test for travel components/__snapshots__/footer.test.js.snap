// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer component test should match snapshot 1`] = `
<div
  className="css-dx7dn0"
>
  <div
    className="chakra-stack css-owjkmg"
  >
    <div
      className="chakra-stack css-k7bpxb"
    >
      <div
        className="chakra-stack css-aju1p1"
      >
        <h3
          className="chakra-heading css-4g6ai3"
          onClick={[Function]}
        >
          About Us
        </h3>
        <h3
          className="chakra-heading css-4g6ai3"
          onClick={[Function]}
        >
          Contact Us
        </h3>
      </div>
      <hr
        aria-orientation="vertical"
        className="chakra-divider css-qj2fhx"
      />
      <div
        className="chakra-stack css-owjkmg"
      >
        <iframe
          allow="autoplay; clipboard-write; encrypted-media; picture-in-picture; web-share"
          allowfullscreen="true"
          frameborder="0"
          height="500"
          scrolling="no"
          src="https://www.facebook.com/plugins/page.php?href=https%3A%2F%2Fwww.facebook.com%2FWonderDiscoveryTravelSriLanka%2F&tabs=messages&width=340&height=500&small_header=false&adapt_container_width=true&hide_cover=false&show_facepile=true&appId"
          style={
            Object {
              "border": "none",
              "overflow": "hidden",
            }
          }
          width="340"
        />
                             
      </div>
    </div>
    <hr
      aria-orientation="horizontal"
      className="chakra-divider css-4wijg1"
    />
  </div>
</div>
`;

exports[`footer2 component test should match snapshot 1`] = `
<div
  className="css-dx7dn0"
>
  <div
    className="chakra-stack css-owjkmg"
  >
    <div
      className="chakra-stack css-k7bpxb"
    >
      <div
        className="chakra-stack css-aju1p1"
      >
        <h3
          className="chakra-heading css-4g6ai3"
          onClick={[Function]}
        >
          About Us
        </h3>
        <h3
          className="chakra-heading css-4g6ai3"
          onClick={[Function]}
        >
          Contact Us
        </h3>
      </div>
      <hr
        aria-orientation="vertical"
        className="chakra-divider css-qj2fhx"
      />
      <div
        className="chakra-stack css-owjkmg"
      >
        <iframe
          allow="autoplay; clipboard-write; encrypted-media; picture-in-picture; web-share"
          allowfullscreen="true"
          frameborder="0"
          height="500"
          scrolling="no"
          src="https://www.facebook.com/plugins/page.php?href=https%3A%2F%2Fwww.facebook.com%2FWonderDiscoveryTravelSriLanka%2F&tabs=messages&width=340&height=500&small_header=false&adapt_container_width=true&hide_cover=false&show_facepile=true&appId"
          style={
            Object {
              "border": "none",
              "overflow": "hidden",
            }
          }
          width="340"
        />
                             
      </div>
    </div>
    <hr
      aria-orientation="horizontal"
      className="chakra-divider css-4wijg1"
    />
  </div>
</div>
`;
